menas/src/main/scala/za/co/absa/enceladus/rest/services/VersionedModelService.scala

```diff
@@ -86,4 +86,11 @@ abstract class VersionedModelService[C <: VersionedModel](versionedMongoReposito
     }
   }
 
+  def validateEntityName(name: String , entityType:String ):String  = {
```

Comment:
Make `protected`

Reply:
`protected` isn't a thing in scala, I think oyu mean package-private, like `private[services]`

Reply:
My bad, I had completely forgotten about `protected` being legit in scala, haven't used it in ages. You're right, it fits well here.

Reply:
then it becomes Inaccessible for DatasetsServiceTest, Unless and until I extend the versionmodelservice. Please suggest.

Reply:
In test you can create a foo class inheriting the base class to make the function accessible.

Reply:
You don't need to extend anything. Just put the test class in the same package.

Reply:
Tests and their respective classes need to have matching packages in `src/` and `test/`

Reply:
Make it `private[services]`, `protected` in scala works differently than it does in java. In Java, protected methods can be accessed by other classes in the same package or subclasses in any pacakage, but this isn’t true in Scala, it's only accessible from subclasses.

Reply:
Unfortunately `private` and `protected` restricts the member within the class not the package. That leaves three options:
1) Once I suggested (and prefer) as the least evil making only the test less nice. `protected` + foo inheritance in test
2) Leave the method `public`. Ok for test, but opens the internals of the class "to the world"
3) Make it `protected` but assign accessibility to certain class for the sake of the test. IMO worst option, because it pollutes the production code with something not needed there.

Reply:
> Make it `private[services]`, `protected` in scala works differently than it does in java. In Java, protected methods can be accessed by other classes in the same package or subclasses in any pacakage, but this isn’t true in Scala, it's only accessible from subclasses.

I perfer @GeorgiChochov suggestion, as it is clean

Reply:
FYI asked my Scala mentor (without telling him my preference) and from the three options he chose the same one as me 😉 But of course it can be a bias caused by his mentoring 😄

Reply:
Thanks David. Please approve the PR.

Reply:
resolving the conversation